language_service/src/display.rs

```diff
@@ -13,21 +13,35 @@ use std::{
 };
 
 pub(crate) struct CodeDisplay<'a> {
-    pub(crate) compilation: &'a Compilation,
+    compilation: &'a Compilation,
+}
+
+#[derive(Copy, Clone)]
+struct HirLookup<'a> {
```

Comment:
Yeah... since they're used together all over this file, I thought I would give it its own name. Not sure if it actually made the code any clearer though :/